docs/getting-started-guides/mesos-docker.md

```diff
@@ -0,0 +1,165 @@
+- [Optional] [Virtual Box](https://www.virtualbox.org/wiki/Downloads)<br/>
+    Required by Boot2Docker and Docker Machine
+- [Optional] [etcd](https://github.com/coreos/etcd)<br/>
+    Only used locally by integration tests, when building without `KUBE_RELEASE_RUN_TESTS=N`.
+
+Note: On Mac, it's possible to install all the above via [Homebrew](http://brew.sh/).<br/>
+Follow any printed instructions after each step to make sure each is configured correctly.
+
+```
+brew install git
+brew install caskroom/cask/brew-cask
+brew cask install virtualbox
+brew install docker
+brew install boot2docker
+boot2docker init
+boot2docker up
+brew install etcd
+```
+
+***TODO***: apt & yum instructions
+
+### Walkthrough
+
+1. Checkout the Kubernetes source
+
+    ```
+    git clone https://github.com/GoogleCloudPlatform/kubernetes
+    cd kubernetes
+    ```
+
+    By default, that will get you the bleeding edge of master branch.
+    You may want a [release branch](https://github.com/GoogleCloudPlatform/kubernetes/releases) instead,
+    if you have trouble with master.
+
+1. Build Kubernetes-Mesos binaries (cross-compiled)
+
+    ```
+    KUBERNETES_CONTRIB=mesos KUBE_RELEASE_RUN_TESTS=N ./build/release.sh
+    ```
+
+    Environment Variables:
+    - `KUBERNETES_CONTRIB=mesos` enables building of the mesos-specific binaries.
+    - `KUBE_RELEASE_RUN_TESTS=N` disables the unit and integration tests that are by default run after the binaries are built.
+
+1. Build docker images
+
+    Test image includes all the dependencies required for running e2e tests.
+
+    ```
+    ./cluster/mesos/docker/test/build.sh
+    ```
+
+    Mesos-Slave image extends the Mesosphere mesos-slave image to include iptables & docker-in-docker.
+
+    ```
+    ./cluster/mesos/docker/mesos-slave/build.sh
+    ```
+
+    Kubernetes-Mesos image includes the compiled linux binaries.
+
+    ```
+    ./cluster/mesos/docker/km/build.sh
+    ```
+
+1. Configure Mesos-Docker Provider
+
+    ```
+    export KUBERNETES_PROVIDER=mesos/docker
+    ```
+
+    ***Resources***
+
+    It's optionally possible to modify the amount of resources the mesos-slaves will offer for Kubernetes to use.
+    To do so, find the `MESOS_RESOURCES` environment variables in `./cluster/mesos/docker/docker-compose.yml` and modify
+    them to your liking. Because mesos-slave resource auto-detection overlaps work when multiple slaves are on the same
+    node, these have to be explicitly configured.
+
+    ***Important***: The default mesos resource may or may not actually be available on your Docker Engine machine.
+    You may have to increase you VM disk, memory, or cpu allocation in VirtualBox,
+    [Docker Machine](https://docs.docker.com/machine/#oracle-virtualbox), or
+    [Boot2Docker](https://ryanfb.github.io/etc/2015/01/28/increasing_boot2docker_allocations_on_os_x.html).
+
+1. Create cluster
+
+    ```
+    ./cluster/kube-up.sh
+    ```
+
+1. Run End-To-End Tests
+
+    ```
+    ./cluster/test-e2e.sh
```

Comment:
`./cluster/test-e2e.sh` is currently bypassing `./hack/e2e.go` (which uses those params) and going directly to `./hack/ginkgo-e2e.sh`.